### DISCUSSION TITLE
How can I set default value inside loop and calculate them?

### DISCUSSION THREAD

**Discussion by ShekhSaifuddin007** [livewire/livewire#2106] [1. Help] [answered]
My scenario is you can see the image

![prob](https://user-images.githubusercontent.com/48278861/100398931-f5a05180-307a-11eb-8db2-839ac728de4f.PNG)

here, how can I set the default value in the `quantity` field? and I want when I change the `quantity` the total amount will be updated how can I achieve this? here you can see my quantity field is an array I catch it like `addComboProductsInTable.{{ $comboKey }}.quantity`  help me, please

### blade

```blade
@foreach ($addComboProductsInTable as $comboKey => $comboProduct)
    <tr>
      <td class="p-1">
         {{ $comboProduct['name'] }}
      </td>
      <td class="p-1">
        <input type="text" wire:model.defer="addComboProductsInTable.{{ $comboKey }}.quantity" class="form-control" placeholder="Qty" style="width: 50%">
        <br>
        <span class="badge badge-primary">{{ $comboProduct['unit']['name'] }}</span>
      </td>
      <td>
        @php($total += $comboProduct['selling_price'])
         $ {{ $comboProduct['selling_price'] }}
      </td>
      <td class="p-1">
          $ 700
      </td>
      <td class="p-1">
        <button type="button"  wire:click.prevent="removeComboProductFromTable({{ $comboKey }})" class="btn btn-xs btn-danger">
           <i class="bx bx-plus rotate-45"></i>
        </button>
      </td>
   </tr>
@endforeach
```

### component

```php
public $addComboProductsInTable = [];
```

**Comment by ShekhSaifuddin007** [comment]
@joshhanley brother can you help me, please

**Comment by joshhanley** [comment]
@ShekhSaifuddin007 how are you adding products to `$addComboProductsInTable`? What I would do is set the quantity to a default value in the array when you assign a product to it.

To calculate your total, you could, just output `{{ addComboProductsInTable[$comboKey].quantity *  $comboProduct['selling_price'] }}` and change your wire:model.defer to wire:model.lazy on your quantity input, so it will update when you change a quantity and tab away from the field

Hope this helps!

**Reply by ShekhSaifuddin007** [reply]
@joshhanley brother, it gives me this error in `IDE`

![error](https://user-images.githubusercontent.com/48278861/100739218-f3cee900-3400-11eb-81bb-1e54fc4f8d93.PNG)

and when I add a product it gives me this `undefined constant addProductsInPurchaseTable`

![error2](https://user-images.githubusercontent.com/48278861/100739331-21b42d80-3401-11eb-867a-0b458927a78a.PNG)

actually, I create a search dropdown, and when I found a product by search I added them to this table by `array_push($this->addProductsInPurchaseTable, $product);`

### my component

```php
public $searchProducts = '';
public $searchProductResults = [];

public $addProductsInPurchaseTable = [];

public function addProductInTable($id)
{
  $product = [
    'product' => Product::query()
                       ->findOrFail($id)
                        ->toArray(),
     'child' => [0]
    ];

   array_push($this->addProductsInPurchaseTable, $product);

   $this->searchProducts = '';
   $this->searchProductResults = [];
}
```

### search markup

```blade
<fieldset class="searching-area">
  <div class="input-group">
    <input type="text"
           @focus="isOpen = true"
           @click.away="isOpen = false"
           wire:model.debounce.500ms="searchProducts"
           class="form-control"
           placeholder="Search by name/SKU or scan barcode"
           aria-describedby="productSearch" autofocus>

    <span wire:loading wire:target="searchProducts"
          class="mr-1 search-combo-spinner spinner-border spinner-border-sm">
    </span>
    <div class="input-group-prepend">
       <span class="input-group-text" id="productSearch">
         <i class="bx bx-search-alt"></i>
       </span>
    </div>
 </div>

 <div x-show="isOpen" class="search-results custom-scrollbar">
   <ul>
      @forelse ($searchProductResults as $productResult)
        <li wire:click.prevent="addProductInTable({{ $productResult->id }})">
         <div>{{ $productResult->name }}</div>
        </li>
     @empty
      @if (strlen($searchProducts) >= 2)
         <li>
             <div>No result found</div>
           </li>
         @endif
       @endforelse
    </ul>
  </div>
</fieldset>
```

### my blade where add the product

```blade
@foreach ($addProductsInPurchaseTable as $purchaseKey => $purchaseProduct)
    <div class="card collapse-header border-left-2 border-right-2" wire:key="purchase_{{ $purchaseKey }}">
      <div class="card-header collapsed border-top border-left-0 border-right-0 border-bottom cursor-default">
        <div class="d-flex align-items-center justify-content-between">
           <div class="align-middle font-size-base ml-30px w-25 cursor-pointer text-truncate" id="posFilterOption_{{ $purchaseKey }}"
                    data-toggle="collapse" role="button" data-target="#posFilter_{{ $purchaseKey }}" aria-expanded="false" aria-controls="posFilter_{{ $purchaseKey }}">
             <span class="text-primary">
                {{ $purchaseProduct['product']['name'] }}
            </span>
          </div>

          <div class="form-label-group position-relative has-icon-left mb-0">
             <input style="width: 100px" type="number" id="Quantity_{{ $purchaseKey }}"
                         wire:model.lazy="addProductsInPurchaseTable.{{ $purchaseKey }}.quantity"
                         class="form-control input-sm" placeholder="Qty">
             <div class="form-control-position">
               <i class="bx bxl-quora" style="top: -2px"></i>
             </div>
              <label for="Quantity_{{ $purchaseKey }}">Quantity</label>
            </div>

           <div class="font-size-base">
              <span>${{ addProductsInPurchaseTable[$purchaseKey].quantity * $purchaseProduct['product']['selling_price'] }}</span>
            <span>
             <a href="#" class="text-danger">
               <i class="bx bx-trash font-size-base mr-semi ml-semi"></i>
             </a>
          </span>
        </div>
      </div>
    </div>
  </div>
@endforeach
```

I want it when I add a new product the input field have by default `value=1` and when updating the input the `total amount` would be updated

**Reply by joshhanley** [reply]
@ShekhSaifuddin007 looks like you have a typo on this line `<span>${{ addProductsInPurchaseTable[$purchaseKey].quantity * $purchaseProduct['product']['selling_price'] }}</span>` 

Change `<span>${{....` to `<span>{{$...`

**Reply by ShekhSaifuddin007** [reply]
@joshhanley same problem with `quantity` `$` this was just dollar sign for the currency

![error3](https://user-images.githubusercontent.com/48278861/100746823-6ba21100-340b-11eb-9c00-866a1a4e61a8.PNG)

when I add product

![error4](https://user-images.githubusercontent.com/48278861/100746847-76f53c80-340b-11eb-82d5-49825bf20f10.PNG)

**Reply by joshhanley** [reply]
@ShekhSaifuddin007 yeah similar issue, make sure you wrap quantity in brackets and quotes. `{{ $addProductsInPurchaseTable[$purchaseKey]['quantity'] .....`

**Reply by ShekhSaifuddin007** [reply]
@joshhanley Like this it shows `Undefined index: quantity` actually my **Product** model has no `quantity` field, this will save in another table, I want to do this for **UI**

**Reply by joshhanley** [reply]
@ShekhSaifuddin007 hmm you probably need to change the $addProductsInPurchaseTable array to a multidimentional array, so when you push to the array:

```
array_push($this->addProductsInPurchaseTable, [
    'product' => $product,
    'quantity' => 0
]);
```

You will need to update the rest your code to use that new structure though. But this line should now work: `wire:model.lazy="addProductsInPurchaseTable.{{ $purchaseKey }}.quantity"` and be populated with 0 to start with.

**Comment by ShekhSaifuddin007** [comment, marked as answer]
@joshhanley yes this work, my code bellow, Thank you brother ❤💙

```php
public function addProductInTable($id)
{
  $product = Product::query()
                    ->findOrFail($id)
                    ->toArray();

 array_push($this->addProductsInPurchaseTable, [
   'product' => $product,
   'quantity' => 1,
   'child' => [0]
 ]);

  $this->searchProducts = '';
  $this->searchProductResults = [];
}
```
